main.js

```diff
@@ -0,0 +1,173 @@
+import { faker } from 'https://cdn.skypack.dev/@faker-js/faker'; //https://fakerjs.dev/api/image.html
+
+
+//object to store state of comments
+const commentsObject = {
+  numberOfComments: 0,
+  comments: {}
+};
+
+//overall container to hold all comments
+const $commentContainer = $('#comment-container');
+
+
+$(document).ready(function() {
+
+  //ANCHOR creates new comment when form is submitted
+  $('#new-comment-form').submit(function(e) {
+    e.preventDefault() //prevents submission from reloading page
+
+    //each comment created adds to the numberOfComments in the commentsObject
+    commentsObject.numberOfComments++
+    const $commentNumber = commentsObject.numberOfComments
+
+    //retrieving values of Name and Message 
+    const $commentName = $(this).find('#comment-name').val()
+    const $commentMessage = $(this).find('#comment-message').val()
+
+    //populates commentObject with data on each comment
+    commentsObject.comments[`comment${$commentNumber}`] = {
```

Comment:
It's a little bit hard to conceptualize what's going on here. If you're intent on making `comments` a piece of data that you loop through, I would make it a simple array and rely on the array length for the number of comments.